src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -717,6 +717,27 @@ public void stringNullAndNullWorksToo() {
         assertEquals("null", null);
     }
 
+    private static class NullToString {
+        @Override
+        public String toString() {
+            return null;
+        }
+    }
+
+    @Test
+    public void nullToString() {
+        try {
+            assertEquals(new NullToString(), new NullToString());
+        } catch (AssertionError e) {
+            assertEquals("expected: org.junit.tests.assertion.AssertionTest$NullToString<null> but "
+                            + "was: org.junit.tests.assertion.AssertionTest$NullToString<null>",
+                    e.getMessage());
+            return;
+        }
+
+        fail("AssertionError expected");
```

Comment:
If we could could do that without hurting readability that is fine. For tests I am fine with some duplication if it makes the code clear.